### PR TITLE
[WIP] Revive the JavaScript test scenarios

### PR DESCRIPTION
**Pull request by adamelso**
I've managed to get the test runner working on Travis for Behat Scenarios tagged with `@javascript`.

Additionally, I fixed one of the scenarios `features/backend/countries.feature:70`, but there quite a few that are broken or failing.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,9 +8,9 @@ php:
     - 5.5
 
 env:
-    - TEST_SUITE=1
+    #- TEST_SUITE=1
```
Comment (adamelso) [inline .travis.yml]: todo: uncomment

File: features/backend/countries.feature [hunk 2]
```diff
@@ -69,7 +69,8 @@ Feature: Countries and provinces
     @javascript
     Scenario: Creating new country with provinces
         Given I am on the country creation page
-         When I fill in "Name" with "Poland"
+         When I click "En_us"
```
Comment (adamelso) [inline features/backend/countries.feature]: Is there a better way to describe or handle this? Normally a locale would be set as the default, but when there is none, the form field for "Name" is not visible until you click on the locale tab.

**Comment by pjedrzejewski** [comment]
Niiice! I will have a look at other scenarios.

File: .travis.yml [hunk 3]
```diff
@@ -40,7 +40,9 @@ before_script:
     - echo "zend.enable_gc = 0" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
     - echo "memory_limit=4096M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
 
-#   - export MINK_EXTENSION_PARAMS='base_url=http://localhost:8000/app_test.php'
+    - >
+        export BEHAT_PARAMS='{ "extensions": { "Behat\\MinkExtension": { "base_url": "http://localhost:8000/app_test.php/" } } }'
```
Comment (stloyd) [inline .travis.yml]: I would set this directly in `behat.yml.dist`.

**Comment by adamelso** [comment]
One bug I found is with this scenario:

```feature
    Scenario: Creating product with boolean attribute
        Given I am on the product creation page
        When I fill in the following:
            | Name        | Manchester United tee   |
            | Description | Interesting description |
            | Price       | 59.99                   |
        And go to "Attributes" tab
        And I click "Add"
        And I select "T-Shirt fare trade" from "Attribute"
        And I check "Value"
        When I press "Create"
        Then I should be on the page of product "Manchester United tee"
        And "Product has been successfully created." should appear on the page
```

The value field does not change to a checkbox when a boolean attribute is selected.

For this other genuinely failing scenarios, shall I create a new issue for each one?

screen shot 2015-04-18 at 16 35 02

**Comment by adamelso** [comment]
Closing in favour of #3055

**Pull request closed without merging**